pkg/alignchecker/alignchecker.go

```diff
@@ -143,3 +145,86 @@ func check(name string, toCheck []reflect.Type, structs map[string]structInfo, c
 
 	return nil
 }
+
+// Read ELF symbols from .rodata and compare the values against given toCheck values
+func valueCheck(pathToObj string, toCheck map[string]reflect.Value) error {
+	f, err := elf.Open(pathToObj)
+	if err != nil {
+		return fmt.Errorf("ELF failed to open %s: %s", pathToObj, err)
+	}
+	defer f.Close()
+
+	elf_syms, err := f.Symbols()
+	if err != nil {
+		return fmt.Errorf("Failed to get ELF symbols")
+	}
+
+	rodata := f.Section(".rodata")
+	if rodata == nil {
+		return fmt.Errorf("Failed to open .rodata section")
+	}
+	rodata_rd := rodata.Open()
+
+	for _, sym := range elf_syms {
+		go_val, check := toCheck[sym.Name]
+		if !check {
+			continue
+		}
+
+		val_size := uint64(go_val.Type().Size())
+		if val_size != sym.Size {
+			return fmt.Errorf("%s: BPF size (%v) differs from Go value size (%v)", sym.Name, sym.Size, val_size)
```

Comment:
This would a great candidate for a sentinel error value, where we create a specific error type representing this condition.

```
type ErrBPFSizeMismatch struct { # Feel free to pick a name you like better
     name string
     bpfSize, valueSize uint64
}

...

return ErrBPFSizeMismatch{
  name: sym.Name,
  bpfSize: sym.Size,
  valueSize: valSize
}
```

Reply:
Thanks, makes sense.